infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsLogEntryReader.java

```diff
@@ -146,6 +158,13 @@ private boolean isValidTransactionEntry(@NonNull OpaqueEntry entry) {
             return false;
         }
 
+        // In DYNAMIC mode, chances are that tables corresponding to some streams to replicate were not opened
+        // when LogReplicationConfig was initialized. So these streams will be missing from the list of streams to
+        // replicate. Check the registry table and add them to the list in that case.
+        if (!streamUUIDs.containsAll(txEntryStreamIds)) {
+            config.syncWithRegistry();
```

Comment:
log an info-level message here that a new stream to replicate was discovered.  Also add the stream's name